sdks/python/apache_beam/transforms/window.py

```diff
@@ -190,12 +192,29 @@ def __init__(self, end):
   def max_timestamp(self):
     return self.end.predecessor()
 
-  def __cmp__(self, other):
+  def cmp(self, other):
     # Order first by endpoint, then arbitrarily.
-    return cmp(self.end, other.end) or cmp(hash(self), hash(other))
+    end_cmp = (self.end > other.end) - (self.end < other.end)
+    hash_cmp = (hash(self) > hash(other)) - (hash(self) < hash(other))
+    return end_cmp or hash_cmp
 
   def __eq__(self, other):
-    raise NotImplementedError
+    return self.cmp(other) == 0
```

Comment:
It doesn't look like this takes care of the case where other is not of type `BoundedWindow`.